sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -374,10 +374,12 @@ private RxDocumentClientImpl(URI serviceEndpoint,
         this.diagnosticsClientConfig.withConsistency(consistencyLevel);
         this.throughputControlEnabled = new AtomicBoolean(false);
 
-        logger.info(
-            "Initializing DocumentClient [{}] with"
-                + " serviceEndpoint [{}], connectionPolicy [{}], consistencyLevel [{}], directModeProtocol [{}]",
-            this.clientId, serviceEndpoint, connectionPolicy, consistencyLevel, configs.getProtocol());
```

Comment:
Do not do potentially costly conversion of object to a string for logging if the info is not going to be
logged anyway. One should not call logger.info/debug/warn unless the right log leve is enabled. The rest of
the logger.info/debug/warn calls in this class are not taking much CPU even if wasted so are left as is.

Reply:
is check info necessary since it's the default setting in most cases?

Reply:
Yes, some apps run in warn level and up only.